File: app/src/main/java/com/za/androidauthenticator/data/contract/SitesIconContract.java
package com.za.androidauthenticator.data.contract;

import com.za.androidauthenticator.R;

import java.util.HashMap;
import java.util.Map;

public class SitesIconContract {
    private static final Map<String, Integer> mapSiteNameWithIcon;

    static {
        mapSiteNameWithIcon = new HashMap<>();
        mapSiteNameWithIcon.put("google", R.drawable.ic_google);
        mapSiteNameWithIcon.put("microsoft", R.drawable.ic_microsoft);
        mapSiteNameWithIcon.put("facebook", R.drawable.ic_facebook);
        mapSiteNameWithIcon.put("twitter", R.drawable.ic_twitter);
    }

    public static Integer getIconId(String siteName) {
        siteName = siteName.toLowerCase();
        return mapSiteNameWithIcon.containsKey(siteName) ? mapSiteNameWithIcon.get(siteName) : R.drawable.ic_web;
    }
}

File: app/src/main/java/com/nhq/authenticator/data/contract/SitesIconContract.kt
package com.nhq.authenticator.data.contract

import com.nhq.authenticator.R
import java.util.*

object SitesIconContract {
    private var mapSiteNameWithIcon = mutableMapOf<String, Int?>()

    init {
        mapSiteNameWithIcon["google"] = R.drawable.ic_google
        mapSiteNameWithIcon["microsoft"] = R.drawable.ic_microsoft
        mapSiteNameWithIcon["facebook"] = R.drawable.ic_facebook
        mapSiteNameWithIcon["twitter"] = R.drawable.ic_twitter
    }

    @JvmStatic
    fun getIconId(siteName: String): Int? {
        var siteName = siteName
        siteName = siteName.toLowerCase(Locale.ROOT)
        return if (mapSiteNameWithIcon.containsKey(siteName)) mapSiteNameWithIcon[siteName]
            else R.drawable.ic_web
    }
}